src/Operations.jl

```diff
@@ -243,22 +243,46 @@ function collect_project!(ctx::Context, pkg::PackageSpec, path::String, fix_deps
     return
 end
 
-istracking(pkg) = pkg.path !== nothing || pkg.repo.source !== nothing
-isfixed(pkg) = istracking(pkg) || pkg.pinned
+is_tracking_path(pkg) = pkg.path !== nothing
+is_tracking_repo(pkg) = pkg.repo.source !== nothing
+is_tracking_registry(pkg) = !is_tracking_path(pkg) && !is_tracking_repo(pkg)
+isfixed(pkg) = !is_tracking_registry(pkg) || pkg.pinned
+
+function collect_developed!(ctx::Context, pkg::PackageSpec, developed::Vector{PackageSpec})
+    source = project_rel_path(ctx, source_path(pkg))
+    Pkg.activate(source) do
```

Comment:
Instead of activating here, can't we give the path to `Context`? Something like `Context(env = EnvCache(source))`.